torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -2404,6 +2404,92 @@ Stmt* TermExpander::mutate(const Block* v) {
   return fuseSyncThreads(new_block);
 }
 
+// SimplifierUnderContext
+
+Stmt* SimplifierUnderContext::mutate(const For* v) {
```

Comment:
Can you explain the overall idea behind this function and what it tries to accomplish?

Reply:
This function tries to record the bounds info of index vars in `for` stmts and those info will be used later when simplifying expressions with the index vars. An example is, 
```
for (i=0; i<5; i++){
  x = (i + 20) % 5;
}
``` 
Without any information about `i`, we are not able to simply the assignment stmt any further; but with the bounds information provided in the `for` stmt, `(i + 20) % 5` can be simplified to `i`. I have seen such stmts in real case. 

Back to your question, the `mutate` function here records the bounds info of `i`.